tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,741 @@
+                      command line (computed using the C<ArchiveLocation> value in table 
+                      C<tarchive> for the given archive).
+  - $filePathsRef: reference to the array that contains the absolute paths of all files found in tables
+                   C<files>, C<files_intermediary>, C<parameter_file>, C<mri_protocol_violated_scans>
+                   C<mri_violations_log> and C<MRICandidateErrors> that are tied to the upload(s) passed
+                   on the command line.
+  - $tarchiveID: ID of the DICOM archive (in table C<tarchive>) associated to the upload(s) passed on the command line.
+                 
+=cut
+sub backupFiles {
+    my($archiveLocation, $filePathsRef, $tarchiveID) = @_;
+    
+    # Put in @files the absolutes paths of all files in tables files, files_intermediary
+    # parameter_file, mri_protocol_violated_scans, mri_violations_log and MRICandidateErrors
+    # that are tied to the upload(s). Also put in @files the archive itself (found in table tarchive).
+    my @files;
+    push(@files, map { $_->{'File'} }          @{ $filePathsRef->{'files'} });
+    push(@files, map { $_->{'File'} }          @{ $filePathsRef->{'files_intermediary'} });
+    push(@files, map { $_->{'Value'} }         @{ $filePathsRef->{'parameter_file'} });
+    push(@files, map { $_->{'minc_location'} } @{ $filePathsRef->{'mri_protocol_violated_scans'} });
+    push(@files, map { $_->{'MincFile'} }      @{ $filePathsRef->{'mri_violations_log'} });
+    push(@files, map { $_->{'MincFile'} }      @{ $filePathsRef->{'MRICandidateErrors'} });
+    push(@files, $archiveLocation);
+    
+    foreach my $f (@files) {
+        # If file does not exist (i.e database and file system are not in sync)
+        if(!-e $f) {
+            if($dieOnFileError) {
+                print STDERR "Cannot backup database files:\n";
+                print STDERR "\tFile $f is in the database but was not found in $dataDirBasepath. Aborting.\n";
+                exit $NeuroDB::ExitCodes::MISSING_FILES;
+            }
+            warn "Warning! File $f was not found in $dataDirBasepath (will not be backed up)\n";
+        } elsif(!-r $f) {
+            if($dieOnFileError) {
+                print STDERR "Cannot backup database files:\n";
+                print STDERR "\tFile $f is in database but is not readable. Aborting.\n";
+                exit $NeuroDB::ExitCodes::UNREADABLE_FILE;
+            }
+            warn "Warning! File $f is not readable (will not be backed up)\n";
+        }
+    }
+    
+    # Create a temporary file that will list the absolute paths of all
+    # files to backup (archive). 
+    my($fh, $tmpFileName) = tempfile("$0.filelistXXXX", UNLINK => 1);
+    foreach(@files) {
+        print $fh "$_\n";
+    }
+    close($fh);
+    
+    # Put all files in a big compressed tar ball
+    my $filesBackupPath = &getBackupFileName($tarchiveID);
+    print "Backing up files related to the upload(s) to delete...\n";
```

Comment:
Would add a "\n" before Backing up files... so that the list of files that will be backed up can stand out of the warnings in the terminal.

Reply:
OK